### PR TITLE
feat: add SafeBigNumber and process price as big number

### PR DESCRIPTION
**Pull request by cehali**

### CODE REVIEW

File: packages/oracle-node/src/numbers/SafeBigNumber.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  add(other: NumberArg): SafeBigNumber {
+    return new SafeBigNumber(this._value.add(other.toString()), this._decimals);
+  }
+
+  sub(other: NumberArg): SafeBigNumber {
+    return new SafeBigNumber(this._value.sub(other.toString()), this._decimals);
+  }
+
+  div(other: NumberArg): SafeBigNumber {
+    return new SafeBigNumber(this._value.div(other.toString()), this._decimals);
```
Comment (koonopek) [inline packages/oracle-node/src/numbers/SafeBigNumber.ts:47]: What happens if we do: `SafeBigNumber.from(1).div(10)` ?

File: packages/oracle-node/test/numbers/SafeBigNumber.spec.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+    describe("div", () => {
+      test("should div", () => {
+        const result = SafeBigNumber.from(100, DEFAULT_DECIMALS).div(100);
+        expect(result).toEqual(SafeBigNumber.from(1, DEFAULT_DECIMALS));
+      });
+
+      test("should return 0 if float < 0", () => {
+        const result = SafeBigNumber.from(10, DEFAULT_DECIMALS).div(100);
+        expect(result).toEqual(SafeBigNumber.from(0, DEFAULT_DECIMALS));
```
Comment (koonopek) [inline packages/oracle-node/test/numbers/SafeBigNumber.spec.ts:109]: Why is this ok?

File: packages/oracle-node/src/numbers/SafeBigNumber.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  abs(): SafeBigNumber {
+    return new SafeBigNumber(this._value.abs(), this._decimals);
+  }
+
+  add(other: NumberArg): SafeBigNumber {
+    return new SafeBigNumber(this._value.add(other.toString()), this._decimals);
+  }
+
+  sub(other: NumberArg): SafeBigNumber {
```
Comment (koonopek) [inline packages/oracle-node/src/numbers/SafeBigNumber.ts:42]: What happens if `other` is float in any case?

File: packages/oracle-node/src/numbers/SafeBigNumber.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  assertNonNegative() {
+    utils.assert(
+      !this._value.isNegative(),
+      `${this._value.toString()} is smaller than 0`
+    );
+  }
+
+  unsafeToNumber() {
+    return this._value.toNumber();
```
Comment (koonopek) [inline packages/oracle-node/src/numbers/SafeBigNumber.ts:82]: Should we divde by `10**this.decimals` here

File: packages/oracle-node/test/numbers/SafeBigNumber.spec.ts (hunk 2 above)
Comment (cehali) [reply packages/oracle-node/test/numbers/SafeBigNumber.spec.ts:109]: It is the same as in BigNumber.js

File: packages/oracle-node/src/numbers/SafeBigNumber.ts (hunk 3 above)
Comment (cehali) [reply packages/oracle-node/src/numbers/SafeBigNumber.ts:42]: It will throw an error, just like BigNumber.js

**Pull request closed without merging**